src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1849,6 +1868,9 @@ private BoundExpression CheckValue(BoundExpression expr, BindValueKind valueKind
                 case BoundKind.OutDeconstructVarPendingInference:
                     Debug.Assert(valueKind == BindValueKind.RefOrOut);
                     return expr;
+                case BoundKind.DiscardedExpression:
+                    Debug.Assert(valueKind == BindValueKind.Assignment);
```

Comment:
After latest commit, we can indeed.

---
In reply to: [88552640](https://github.com/dotnet/roslyn/pull/15192#discussion_r88552640) [](ancestors = 88552640)